app/src/components/layout/Navbar/Navbar.tsx

```diff
@@ -25,30 +26,39 @@ import { ConfigurableLogo, DefaultNavLogo } from '../LogoComponents';
 
 const navItems = [
   {
-    title: 'home',
+    title: 'Home',
     path: '/',
     key: 'home',
   },
   {
-    title: 'blocks',
+    title: 'Blocks',
     path: '/blocks',
     key: 'blocks',
   },
   {
-    title: 'peers',
+    title: 'Peers',
     path: '/peers',
     key: 'peers',
   },
   {
-    title: 'validators',
+    title: 'Validators',
     path: '/validators',
     key: 'validators',
   },
 ];
 
 export const Navbar: React.FC = () => {
+  const [isOpened, setIsOpened] = useState<boolean>();
+
+  const [selectedRoute, setSelectedRoute] = useState(null);
+
+  const handleNavItemSelection = (event: {
```

Comment:
We should always have access to a  `key` or `id` in the individual `NavItem` component and pass that to this method, then have that key set the `selectedRoute`. We should never need access to something like `event.target.innerText` unless absolutely necessary.

Reply:
I'll definitely take the `id` route if all else fails.